experanto/interpolators.py

```diff
@@ -427,3 +433,86 @@ def __init__(self, file_name, data) -> None:
 
     def get_data(self) -> np.array:
         return np.full((1,) + self.image_size, self.interleave_value)
+
+
+class SpikesInterpolator(Interpolator):
+    def __init__(
+        self,
+        root_folder: str,
+        sampling_rate: float = 100.0,  # Default to 100 Hz
+        normalize: bool = False,
+        **kwargs,
+    ) -> None:
+        """
+        Interpolator specifically designed for spike train data.
+
+        Args:
+            root_folder (str): Path to the root folder containing spike data.
+            sampling_rate (float): Sampling rate in Hz (default: 100 Hz).
+            normalize (bool): Whether to normalize spike counts.
+        """
+        super().__init__(root_folder)
+        meta = self.load_meta()
+        self.sampling_rate = meta.get("sampling_rate", sampling_rate)
+        self.time_delta = 1.0 / self.sampling_rate
+        self.start_time = meta["start_time"]
+        self.end_time = meta["end_time"]
+        self.valid_interval = TimeInterval(self.start_time, self.end_time)
+
+        self.n_units = meta["n_signals"]  # Number of recorded spike units
+        self.normalize = normalize
+
+        # Load spike data from .npy or .mem file
+        if (self.root_folder / "spikes.npy").exists():
+            self._data = np.load(self.root_folder / "spikes.npy")
+        else:
+            self._data = np.memmap(
+                self.root_folder / "data.mem",
+                dtype=meta["dtype"],
+                mode="r",
+                shape=(meta["n_timestamps"], self.n_units),
+            )
+
+        # Load normalization parameters if needed
+        if self.normalize:
+            self._init_normalization()
+
+    def _init_normalization(self):
+        """Load normalization parameters (mean & std) for spike data."""
+        self.mean = np.load(self.root_folder / "meta/means.npy")
+        self.std = np.load(self.root_folder / "meta/stds.npy")
+        self.mean = self.mean.T
+        self.std = self.std.T
+        self._precision = 1 / self.std
+
+    def interpolate(self, times: np.ndarray) -> tuple[torch.Tensor, np.ndarray]:
+        """
+        Interpolates spike data using binned counts.
+
+        Args:
+            times (np.ndarray): Query timestamps.
+
+        Returns:
+            tuple:
+                - torch.Tensor: Spike counts per bin.
+                - np.ndarray: Boolean mask indicating valid times.
+        """
+        valid = self.valid_times(times)
+        valid_times = times[valid]
+
+        # Convert times to indices in spike data
+        idx = np.round(valid_times * self.sampling_rate).astype(int)
+
+        # Extract spike count data for the interval
+        interval_res = self._data[idx]
+
+        # Compute spike counts per bin (to match old binned counts format) (trapezoidal integration)
+        spike_counts = torch.tensor([
+            np.trapz(each[2:], dx=1 / self.sampling_rate, axis=0) * 1000
```

Comment:
`1000` should not be a hardcoded value but rather a parameter in the init properly names and such that it could be changed later:)